### PR TITLE
Updated the return type section of to_datetime 

### LINKED ISSUE #42229: BUG: to_datetime() returns object instead of datetime type or raising exception
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example

```python
print(pd.to_datetime(['2021-03-14 12:05:45 -0400', '2021-03-14 03:06:00 -0400']))
print(pd.to_datetime(['2021-03-14 01:06:30 -0500', '2021-03-13 23:23:40 -0500']))
print(pd.to_datetime(['2021-03-14 03:06:00 -0400', '2021-03-14 01:06:30 -0500'], errors='raise'))
print(pd.to_datetime(['2021-03-14 03:06:00 -0400', '2021-03-14 01:06:30 -0500'], errors='coerce'))
print(pd.to_datetime(['2021-03-14 03:06:00 -0400', '2021-03-14 01:06:30 -0500'], errors='ignore'))
```
The output is
```
DatetimeIndex(['2021-03-14 12:05:45-04:00', '2021-03-14 03:06:00-04:00'], dtype='datetime64[ns, pytz.FixedOffset(-240)]', freq=None)
DatetimeIndex(['2021-03-14 01:06:30-05:00', '2021-03-13 23:23:40-05:00'], dtype='datetime64[ns, pytz.FixedOffset(-300)]', freq=None)
Index([2021-03-14 03:06:00-04:00, 2021-03-14 01:06:30-05:00], dtype='object')
Index([2021-03-14 03:06:00-04:00, 2021-03-14 01:06:30-05:00], dtype='object')
Index([2021-03-14 03:06:00-04:00, 2021-03-14 01:06:30-05:00], dtype='object')
```
#### Problem description

The first two lines work as expected, but the following three lines do not return a `DatetimeIndex` but also don't raise any exceptions in the process. This behavior is undocumented, and quite hard to debug/track down, since the failure is completely silent. It seems that the reason is the input contains times from two different time zones.

#### Expected Output
Either a DatetimeIndex object or an exception to be raised.
 
#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : 7c48ff4409c622c582c56a5702373f726de08e96
python           : 3.9.1.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 20.5.0
Version          : Darwin Kernel Version 20.5.0: Sat May  8 05:10:33 PDT 2021; root:xnu-7195.121.3~9/RELEASE_X86_64
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.2.5
numpy            : 1.19.5
pytz             : 2020.5
dateutil         : 2.8.1
pip              : 21.1.2
setuptools       : 49.2.1
Cython           : None
pytest           : 6.2.4
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : 2.8.6 (dt dec pq3 ext lo64)
jinja2           : 2.11.2
IPython          : 7.22.0
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : 3.3.3
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pyxlsb           : None
s3fs             : None
scipy            : 1.6.0
sqlalchemy       : 1.4.2
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
numba            : None

</details>

### PR DESCRIPTION
**Pull request by prabha-git**
Updated the return type section of to_datetime to include list-like mixed timezone inputs

- [x] closes #42229
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry - Updated the return type section of to_datetime to include list-like mixed timezone inputs

### CODE REVIEW

File: pandas/core/tools/datetimes.py [hunk 1]
```diff
@@ -765,6 +765,7 @@ def to_datetime(
         - list-like: DatetimeIndex
         - Series: Series of datetime64 dtype
         - scalar: Timestamp
+        - list-like with mixed timezones: Index of object
```
Comment (mroeschke) [inline pandas/core/tools/datetimes.py:768]: `Index of object dtype`

Comment (prabha-git) [reply pandas/core/tools/datetimes.py]: Updated

File: pandas/core/tools/datetimes.py [hunk 2]
```diff
@@ -765,6 +765,7 @@ def to_datetime(
         - list-like: DatetimeIndex
         - Series: Series of datetime64 dtype
         - scalar: Timestamp
+        - list-like with mixed timezones: Index of object dtype
```
Comment (mroeschke) [inline pandas/core/tools/datetimes.py:768]: Actually on second thought might be better to combine with `list-like: DatetimeIndex` like

```
- Index:
    - DatetimeIndex if timezone naive or aware with the same timezone
    - Index of object dtype if timezone aware with mixed timezones
```

**Comment by prabha-git** [comment]
@mroeschke - updated as per you suggestion

File: pandas/core/tools/datetimes.py [hunk 3]
```diff
@@ -762,10 +762,13 @@ def to_datetime(
...
         Return type depends on input:
 
-        - list-like: DatetimeIndex
+        - list-like:
+            - DatetimeIndex, if timezone naive or aware with the same timezone
+            - Index of object dtype, if timezone aware with mixed timezones
         - Series: Series of datetime64 dtype
         - scalar: Timestamp
 
+
```
Comment (mroeschke) [inline pandas/core/tools/datetimes.py:771]: Nit: Could you remove this newline?

**Review by mroeschke — COMMENTED** [review]
One small comment otherwise LGTM

File: pandas/core/tools/datetimes.py (hunk 3 above)
Comment (prabha-git) [reply pandas/core/tools/datetimes.py]: Yes, I can

**Comment by prabha-git** [comment]
Removed the additional line , Thanks for patience and feedback @mroeschke

**Review by mroeschke — APPROVED** [review]

File: pandas/core/tools/datetimes.py [hunk 4]
```diff
@@ -762,7 +762,9 @@ def to_datetime(
         If parsing succeeded.
         Return type depends on input:
 
-        - list-like: DatetimeIndex
+        - list-like:
+            - DatetimeIndex, if timezone naive or aware with the same timezone
```
Comment (smarie) [inline pandas/core/tools/datetimes.py:766]: As discussed in https://github.com/pandas-dev/pandas/issues/42229#issuecomment-870009579

```suggestion
            - DatetimeIndex, if timezone naive or aware with the same time offset.
```

File: pandas/core/tools/datetimes.py [hunk 5]
```diff
@@ -762,7 +762,9 @@ def to_datetime(
...
+            - Index of object dtype, if timezone aware with mixed timezones
```
Comment (smarie) [inline pandas/core/tools/datetimes.py:767]:
```suggestion
            - Index of object dtype, if timezone aware with mixed time offsets
```

**Comment by mroeschke** [comment]
Thanks @prabha-git

**Comment by smarie** [comment]
Hi @prabha-git and @mroeschke ! It seems that the resulting comment is half/half (half speaking about timezone, half speaking about time offset...). Therefore I created https://github.com/pandas-dev/pandas/pull/42494 to complete this PR.

**Pull request merged**